File: src/com/kingmang/lazurite/patterns/VariablePattern.java
package com.kingmang.lazurite.patterns;

import lombok.Getter;

import java.io.Serializable;

public class VariablePattern extends Pattern implements Serializable {

    @Getter
    private String variable;

    public VariablePattern(String variable) {
        this.variable = variable;
    }

    @Override
    public String toString() {
        return variable;
    }
}

File: src/main/java/com/kingmang/lazurite/patterns/VariablePattern.kt
package com.kingmang.lazurite.patterns

import java.io.Serializable

class VariablePattern : Pattern, Serializable  {


    private val variable: String?

    constructor(variable : String) {
        this.variable = variable
    }

    fun getVariable() : String? {
        return variable
    }

    override fun toString(): String {
        return variable!!
    }
}
